### PR TITLE
Initialize project

### PR DESCRIPTION
**Pull request by yhmo**
Signed-off-by: yhmo <yihua.mo@zilliz.com>

#1
#3 
#4

### CODE REVIEW

File: scripts/proto_gen.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
```
Comment (matrixji) [inline scripts/proto_gen.sh:1]: I may introduce a cmake way to generate pb.cc from .proto file, after this is merged.

Comment (yhmo) [reply scripts/proto_gen.sh:1]: Ok

**Pull request merged**